pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -201,13 +215,255 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) error {
-	return fmt.Errorf("not implemented")
+	objectType, secretName := getObjType(esv1beta1.ExternalSecretDataRemoteRef{Key: remoteRef.GetRemoteKey()})
+	switch objectType {
+	case defaultObjType:
+		return a.deleteKeyVaultSecret(ctx, secretName)
+	case objectTypeCert:
+		return a.deleteKeyVaultCertificate(ctx, secretName)
+	case objectTypeKey:
+		return a.deleteKeyVaultKey(ctx, secretName)
+	default:
+		return fmt.Errorf("secret type '%v' is not supported", objectType)
+	}
+}
+
+func getCertificateFromValue(value []byte) (*x509.Certificate, error) {
+	_, localCert, err := pkcs12.Decode(value, "")
+	if err != nil {
+		pemBlock, _ := pem.Decode(value)
+		if pemBlock == nil {
+			return x509.ParseCertificate(value)
+		}
+		return x509.ParseCertificate(pemBlock.Bytes)
+	}
+	return localCert, err
+}
+
+func getKeyFromValue(value []byte) (interface{}, error) {
+	val := value
+	pemBlock, _ := pem.Decode(value)
+	// if a private key regular expression doesn't match, we should consider this key to be symmetric
+	if pemBlock == nil {
+		return val, nil
+	}
+	val = pemBlock.Bytes
+	switch pemBlock.Type {
+	case "PRIVATE KEY":
+		return x509.ParsePKCS8PrivateKey(val)
+	case "RSA PRIVATE KEY":
+		return x509.ParsePKCS1PrivateKey(val)
+	case "EC PRIVATE KEY":
+		return x509.ParseECPrivateKey(val)
+	default:
+		return nil, fmt.Errorf("key type %v is not supported", pemBlock.Type)
+	}
+}
+
+func (a *Azure) setKeyVaultSecret(ctx context.Context, secretName string, value []byte) error {
+	secret, err := a.baseClient.GetSecret(ctx, *a.provider.VaultURL, secretName, "")
+	aerr := &autorest.DetailedError{}
+	conv := errors.As(err, aerr)
+	if err != nil && !conv {
+		return fmt.Errorf("could not get secret %v: could not parse error: %w", secretName, err)
+	}
+	if conv && aerr.StatusCode != 404 {
+		return fmt.Errorf("could not get secret %v: %w", secretName, err)
+	}
+	if err == nil {
+		manager, ok := secret.Tags["managed-by"]
+		if !ok || manager == nil || *manager != managerLabel {
+			return fmt.Errorf("secret %v is not managed by external-secrets", secretName)
+		}
+		val := secret.Value
+		if val != nil && *val == string(value) {
+			return nil
+		}
+	}
+	val := string(value)
+	secretParams := keyvault.SecretSetParameters{
+		Value: &val,
+		Tags: map[string]*string{
+			"managed-by": pointer.String(managerLabel),
+		},
+		SecretAttributes: &keyvault.SecretAttributes{
+			Enabled: pointer.Bool(true),
+		},
+	}
+	_, err = a.baseClient.SetSecret(ctx, *a.provider.VaultURL, secretName, secretParams)
+	if err != nil {
+		return fmt.Errorf("could not set secret %v: %w", secretName, err)
+	}
+	return nil
+}
+
+func (a *Azure) setKeyVaultCertificate(ctx context.Context, secretName string, value []byte) error {
+	val := b64.StdEncoding.EncodeToString(value)
+	localCert, err := getCertificateFromValue(value)
+	if err != nil {
+		return fmt.Errorf("value from secret is not a valid certificate: %w", err)
+	}
+	b := sha1.Sum(localCert.Raw) //nolint
```

Comment:
the x509thumbprint available in Key Vault certificate is a sha1-based (which is bad). Still trying to figure out for alternatives here.

Reply:
sha1 is only a problem if we use it to verify certificate legitimacy. The worst (also, very unlikely) that could happen is that there is a hash collision and eso does not update the remote certificate because it thinks it has been updated already.

Alternatively we could:
* parse the remote `.Cer` and use the x509.SerialNumber to compare local and remote
* use a different algo to compute a hash over the `.Cer` (DER) bytes

Reply:
Went through the "use different algo" approach :) Just calculating with SHA512 should be enough 😄